File: app/src/main/java/xyz/godi/popularmovies/data/MovieDAO.java
package xyz.godi.popularmovies.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import xyz.godi.popularmovies.model.Movie;

@Dao
public interface MovieDAO {

    // get all movies
    @Query("SELECT * FROM Movie")
    LiveData<List<Movie>> getAll();

    // insert movie into the database
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Movie movie);

    @Query("SELECT * FROM Movie Where id = :id")
    Movie getMovieById(int id);

    @Delete
    void delete(Movie movie);
}

File: app/src/main/java/xyz/godi/popularmovies/data/MovieDao.kt
package xyz.godi.popularmovies.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import xyz.godi.popularmovies.model.Movie

/**
 * Created by Farouk on 12/06/20.
 */
@Dao
interface MovieDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertMovieList(movieList: List<Movie>)

    @Query("SELECT * FROM Movie WHERE page = :page_")
    fun getMovieList(page_: Int): List<Movie>
}